dpdata/customsystem.py

```diff
@@ -0,0 +1,436 @@
+                new_data['atom_numbs'] =  atomNumbs
+                continue
+            if key == "coords":
+                continue
+            new_data[key]= deepcopy(data[key])
+        
+        
+        nframes = self.get_nframes()
+        for to_frame in range(nframes):
+            if at_is_cartesian:
+                cart_at = deepcopy(at)
+            else:
+                cart_at = np.array([0.0, 0.0, 0.0])
+                cart_at[0] = at[0]*data['cells'][to_frame][0][0] + at[1]*data['cells'][to_frame][1][0] + at[2]*data['cells'][to_frame][2][0]
+                cart_at[1] = at[0]*data['cells'][to_frame][0][1] + at[1]*data['cells'][to_frame][1][1] + at[2]*data['cells'][to_frame][2][1]
+                cart_at[2] = at[0]*data['cells'][to_frame][0][2] + at[1]*data['cells'][to_frame][1][2] + at[2]*data['cells'][to_frame][2][2]
+            center = mol.center_of_mass
+            newcoords_mol = np.copy(mol.cart_coords) - center + cart_at
+            if "coords" in new_data:
+                new_data["coords"].append(np.concatenate((np.copy(data["coords"][to_frame]), np.array(newcoords_mol))))
+            else:
+                new_data["coords"] = np.concatenate((np.copy(data["coords"][to_frame]), np.array(newcoords_mol))).reshape(1, -1, 3)
+        return new_data
+        
+        
+        
+def Extract_elem(mol):
+    return list(str(site.species.elements[0]) for site in mol.sites)
+
+def assignIdxElem(elem_mol, elem_latt, atomNumbs_latt):
+    elemIdx_mol = list()
+    orderedUniqElem = deepcopy(elem_latt)
+    atomNumbs = deepcopy(atomNumbs_latt)
+    for item in elem_mol:
+        if item in orderedUniqElem:
+            idx = orderedUniqElem.index(item)
+            atomNumbs[idx] += 1
+            elemIdx_mol.append(idx)
+        else:
+            idx = len(orderedUniqElem)
+            elemIdx_mol.append( idx)
+            orderedUniqElem.append(item)
+            atomNumbs.append(1)
+    return list(elemIdx_mol), orderedUniqElem, list(atomNumbs)
+    
+def removeElem(elem_mol, elem, atomNumbs):
+    assert(len(elem) == len(atomNumbs))
+    elem_latt = deepcopy(elem)
+    atomNumbs_latt = deepcopy(atomNumbs)
+    for idx in range(len(elem)):
+        item = elem[idx]
+        if item in elem_mol:
+            atomNumbs_latt[idx] -= 1
+    pickidces = np.where(atomNumbs_latt >= 0)
+    return elem_latt[pickidces], atomNumbs_latt[pickidces], 
+
+
+def c2d(cart, cell):
+    reccell = np.zeros((3,3))
+    Vcell = np.dot(cell[0], np.cross(cell[1], cell[2]))
+    reccell[0] = np.cross(cell[1], cell[2])/Vcell
+    reccell[1] = np.cross(cell[2], cell[0])/Vcell
+    reccell[2] = np.cross(cell[0], cell[1])/Vcell
+    direct = np.zeros(3)
+    direct[0] = cart[0]*reccell[0][0] + cart[1]*reccell[1][0] + cart[2]*reccell[2][0]
+    direct[1] = cart[1]*reccell[0][1] + cart[1]*reccell[1][1] + cart[2]*reccell[2][1]
+    direct[2] = cart[2]*reccell[0][2] + cart[1]*reccell[1][2] + cart[2]*reccell[2][2]
```

Comment:
I think it's equal to
```py
    direct = (reccell.T @ cart).T
```
or something similar. Anyway, you don't need to write for every element...

Reply:
Same comment for `_merge_data`

Reply:
Thank you !!!